### PR TITLE
Sync in a separate thread

### LINKED ISSUE #45: `wallet.sync()` is blocking execution on android
Combined with issue https://github.com/LtbLightning/bdk-rn/issues/44, this becomes quickly problematic.

Performing a `wallet.sync()` on app init result in at least 10 seconds waiting time.

Any other `wallet.sync()` during the session makes the app unresponsive

### PR DESCRIPTION
**Pull request by BitcoinZavior**
Create a new thread on the native side for Sync method
Fixes #45

### CODE REVIEW

**Review by Czino — CHANGES_REQUESTED** [review]
Just tested this on a physical android device and the javascript side is still blocked while syncing :(

File: android/src/main/java/io/ltbl/bdkrn/BdkRnModule.kt [hunk 1]
```diff
@@ -1,6 +1,7 @@
 package io.ltbl.bdkrn
 
 import com.facebook.react.bridge.*
+import com.facebook.react.bridge.UiThreadUtil.runOnUiThread
```
Comment (Czino) [inline android/src/main/java/io/ltbl/bdkrn/BdkRnModule.kt:4]:
```suggestion
```

File: android/src/main/java/io/ltbl/bdkrn/BdkRnModule.kt [hunk 2]
```diff
@@ -393,8 +394,12 @@ class BdkRnModule(reactContext: ReactApplicationContext) :
...
     fun sync(id: String, blockChainId: String, result: Promise) {
         try {
-            getWalletById(id).sync(getBlockchainById(blockChainId), BdkProgress)
-            result.resolve(true)
+            Thread {
+                runOnUiThread {
+                    getWalletById(id).sync(getBlockchainById(blockChainId), BdkProgress)
+                    result.resolve(true)
+                }
+            }.start()
```
Comment (Czino) [inline android/src/main/java/io/ltbl/bdkrn/BdkRnModule.kt:402]: Found that all is needed is not to run it on the `UiThread`

```suggestion
            Thread {
                getWalletById(id).sync(getBlockchainById(blockChainId), BdkProgress)
                result.resolve(true)
            }.start()
```

**Review by Czino — CHANGES_REQUESTED** [review]

File: android/src/main/java/io/ltbl/bdkrn/BdkRnModule.kt (hunk 2 above)
Comment (BitcoinZavior) [reply android/src/main/java/io/ltbl/bdkrn/BdkRnModule.kt:402]: Yes, reached a similar conclusion. Does this resolve the UI thread issue for you?

Comment (Czino) [reply android/src/main/java/io/ltbl/bdkrn/BdkRnModule.kt:402]: Yes!

**Pull request merged**